packages/app/cypress/e2e/integration/settings.spec.ts

```diff
@@ -1,99 +1,94 @@
-    cy.wait('@SetPreferred').its('request.body.variables.value').should('include', '/usr/local/bin/vim')
+      // navigate away and come back
+      // preferred editor selected from dropdown should have been persisted
+      cy.visitApp()
+      cy.get('[href="#/settings"]').click()
+      cy.wait(100)
+      cy.get('[data-cy="Device Settings"]').click()
 
-    cy.contains('Choose your editor...').click()
-    cy.contains('Well known editor').click()
-    cy.wait('@SetPreferred').its('request.body.variables.value').should('include', '/usr/bin/well-known')
+      cy.get('[data-cy="use-well-known-editor"]').should('be.checked')
+      cy.get('[data-cy="use-custom-editor"]').should('not.be.checked')
+    })
 
-    // navigate away and come back
-    // preferred editor selected from dropdown should have been persisted
-    cy.visitApp()
-    cy.get('[href="#/settings"]').click()
-    cy.wait(100)
-    cy.get('[data-cy="Device Settings"]').click()
+    it('allows custom editor', () => {
+      cy.intercept('POST', 'mutation-ExternalEditorSettings_SetPreferredEditorBinary').as('SetPreferred')
 
-    cy.get('[data-cy="use-well-known-editor"]').should('be.checked')
-    cy.get('[data-cy="use-custom-editor"]').should('not.be.checked')
+      // doing invoke instead of `type` since `type` enters keys on-by-one, triggering a mutation
+      // for each keystroke, making it hard to intercept **only** the final request, which I want to
+      // assert contains `/usr/local/bin/vim'
+      cy.findByPlaceholderText('Custom path...').clear().invoke('val', '/usr/local/bin/vim').trigger('input').trigger('change')
+      cy.wait('@SetPreferred').its('request.body.variables.value').should('include', '/usr/local/bin/vim')
 
-    // change to custom editor using input
-    cy.findByPlaceholderText('Custom path...').clear().invoke('val', '/usr/local/bin/vim').trigger('input').trigger('change')
+      // navigate away and come back
+      // preferred editor entered from input should have been persisted
+      cy.get('[href="#/settings"]').click()
+      cy.wait(100)
+      cy.get('[data-cy="Device Settings"]').click()
 
-    // navigate away and come back
-    // preferred editor entered from input should have been persisted
-    cy.get('[href="#/settings"]').click()
-    cy.wait(100)
-    cy.get('[data-cy="Device Settings"]').click()
+      cy.get('[data-cy="use-well-known-editor"]').should('not.be.checked')
+      cy.get('[data-cy="use-custom-editor"]').should('be.checked')
+      cy.get('[data-cy="custom-editor"]').should('have.value', '/usr/local/bin/vim')
+    })
+
+    it('lists file browser as available editor', () => {
+      cy.intercept('POST', 'mutation-ExternalEditorSettings_SetPreferredEditorBinary').as('SetPreferred')
 
-    cy.get('[data-cy="use-well-known-editor"]').should('not.be.checked')
-    cy.get('[data-cy="use-custom-editor"]').should('be.checked')
+      cy.contains('Choose your editor...').click()
+      cy.contains('File System').click()
```

Comment:
This fails on my mac since `Finder` is the equivalent of `File System` (I'm guessing File System is Linux specific?). Should we even check against this one?

Reply:
Ah, good catch. I've added a test-id to our Select element so that we can select items by value rather than by string and switched to using that.